store/storetest/file_info_store.go

```diff
@@ -810,51 +809,55 @@ func testFileInfoGetStorageUsage(t *testing.T, ss store.Store) {
 }
 
 func testGetUptoNSizeFileTime(t *testing.T, ss store.Store) {
-	t.Skip("MM-48627")
 	_, err := ss.FileInfo().GetUptoNSizeFileTime(0)
 	assert.Error(t, err)
 	_, err = ss.FileInfo().GetUptoNSizeFileTime(-1)
 	assert.Error(t, err)
 
-	_, err = ss.FileInfo().PermanentDeleteBatch(model.GetMillis(), 100000)
```

Comment:
I can't reproduce the test failure exactly like in the traceback linked in the JIRA ticket.
I was able to repro it on my local machine by running the test multiple times, using the go test's `count` flag

```
go test -count 2 -v -run "TestFileInfoStore/.*/GetUptoNSizeFileTime" ./store/sqlstore/...
```

This fails consistently with a similar error as in the ticket, because there is an issue with test clean up.

In the quoted line we delete all files made before now. Later in the test we create files in the future. When running the test multiple times with the `count` flag, each test run happens very quickly. So even on the second run, there are files in the future from the first run. These files don't get cleaned up the `PermanentDeleteBatch`, so they appear unexpectedly in the test results.

To fix this, I've

1. Deleted files in the future as well. 16 minutes in the future is arbitrary
2. New files are created in the past

The failures in CI are happening because test file info entries not getting cleaned up because of similar issues. 
I'm hoping both of these fixes will fix the flaky behaviour in the Circle CI as well.

I've also cleaned up some of the code, added comments and better variable names (imho)

Reply:
Huh, I'm unable to reproduce the error, with your command or using `-race`. Could you copy here the error you're getting? I'm curious.

> To fix this, I've
> 
>     1. Deleted files in the future as well. 16 minutes in the future is arbitrary.
> 
>     2. New files are created in the past, to make clean-up behaviour more predictable.

Wouldn't it be enough to create files in the past? Maybe I'm misunderstanding something.

Reply:
What I don't understand is how the tests are conflicting with each other. That shouldn't happen, so that's what we should probably address. Are we sure that in CI we're running more than one test and that this condition is happening?

Reply:
Ugh

Reply:
Ugh